examples/schema/README.md

```diff
@@ -0,0 +1,11 @@
+# Schema tracking and validation
+
+This examples showcases the `SchemaValidator()` adapter. At runtime, it will collect the schema of DataFrame-like objects and stored them as `pyarrow.Schema` objects. On subsequent runs, it will check if the schema of a noed output matches the stored reference schema.
+
+> NOTE­. `SchemaValidator` is currently under `hamilton.experimental.h_schema`. We're currently iterating over the API and are looking forward to integrate it in the main library.
+
+## Content
+- `dataflow.py` defines a dataflow with 3 nodes returning pyarrow, ibis, and pandas dataframes and use the `SchemaValidator()` to track their schema. After running it once, you can change the `Driver` config to use `version="2"` and see the schema validation fail.
```

Comment:
This is an interesting point -- changing the `config` might actually cause a desired change to the schema version... But for now I think this makes sense.

Reply:
It's very use case dependent. Maybe `@config.when()` changes imputation methods, but both should ensure a schema with no nullable type.

I'm currently storing the `HamiltonNode` metadata, including the `.version` field on the `pyarrow.Schema` object. You can enable metadata comparison as part of the schema diff / equality check  via `check_schema_metadata=True`. 

Still have to figure out the best user-facing API though

Reply:
Yeah, it's interesting.  Tangential though.